Source/WarpX.cpp

```diff
@@ -1327,10 +1334,15 @@ WarpX::ReadParameters ()
             }
         }
 
-        // Fill guard cells with backward FFTs if Vay current deposition is used
-        if (WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay)
+        // Without periodic single box, fill guard cells with backward FFTs,
+        // with current correction or Vay deposition
+        if (fft_periodic_single_box == false)
         {
-            WarpX::fill_guards = amrex::IntVect(1);
+            if (current_correction ||
+                current_deposition_algo == CurrentDepositionAlgo::Vay)
+            {
+                fill_guards = amrex::IntVect(1);
```

Comment:
Yes, I believe it also affects the inverse FFT of E and B, because the flag `fill_guards` is passed to the constructor of the spectral solver, and from there to the relevant methods of the class `SpectralFieldData`. We can discuss offline if we want to fill the guard cells only for J by adding more control on this.

Reply:
Thanks to #3302, the new behavior with current correction affects only J.